# GeoEnrichment

GeoEnrichment provides the ability to 
* get facts about a location or area. 
* information about the people, places, and businesses 
 * in a specific area or 
 * within a certain distance or drive time from a location.
* large collection of data sets including population, income, housing, consumer behavior, and the natural environment.
* Site analysis is a popular application

In [1]:
from arcgis.gis import GIS
from arcgis.geoenrichment import *

gis = GIS('https://www.arcgis.com', 'arcgis_python', 'P@ssword123')

## GeoEnrichment coverage

In [2]:
countries = get_countries()
print("Number of countries for which GeoEnrichment data is available: " + str(len(countries)))

#print a few countries for a sample
countries[0:10]

Number of countries for which GeoEnrichment data is available: 137


[<Country name:Albania>,
 <Country name:Algeria>,
 <Country name:Andorra>,
 <Country name:Angola>,
 <Country name:Argentina>,
 <Country name:Armenia>,
 <Country name:Aruba>,
 <Country name:Australia>,
 <Country name:Austria>,
 <Country name:Azerbaijan>]

### Filtering countries by properties

In [3]:
[c.properties.name for c in countries if c.properties.continent == 'Australia']

['Australia',
 'French Polynesia',
 'Indonesia',
 'Malaysia',
 'New Caledonia',
 'New Zealand',
 'Philippines']

## Discovering information for a country
* Data collections, 
* Sub-geographies and 
* Available reports for a country

In [5]:
usa = Country.get('US')

Commonly used properties for the country are accessible using `Country.properties`.

In [6]:
usa.properties.name

'United States'

### Data collections and analysis variables

In [7]:
df = usa.data_collections

# print a few rows of the DataFrame
df.head()

analysisVariable                   alias  \
dataCollectionID                                                    
1yearincrements   1yearincrements.AGE0_CY  2017 Population Age <1   
1yearincrements   1yearincrements.AGE1_CY   2017 Population Age 1   
1yearincrements   1yearincrements.AGE2_CY   2017 Population Age 2   
1yearincrements   1yearincrements.AGE3_CY   2017 Population Age 3   
1yearincrements   1yearincrements.AGE4_CY   2017 Population Age 4   

                                       fieldCategory vintage  
dataCollectionID                                              
1yearincrements   2017 Age: 1 Year Increments (Esri)    2017  
1yearincrements   2017 Age: 1 Year Increments (Esri)    2017  
1yearincrements   2017 Age: 1 Year Increments (Esri)    2017  
1yearincrements   2017 Age: 1 Year Increments (Esri)    2017  
1yearincrements   2017 Age: 1 Year Increments (Esri)    2017

In [8]:
# call the shape property to get the total number of rows and columns
df.shape

(14609, 4)

In [49]:
# get all the unique data collections available
len(df.index.unique())

141

Query the `Age` data collection and get all the unique `analysisVariable`s under that collection

In [10]:
df.loc['Age']['analysisVariable'].unique()

array(['Age.MALE0', 'Age.MALE5', 'Age.MALE10', 'Age.MALE15', 'Age.MALE20',
       'Age.MALE25', 'Age.MALE30', 'Age.MALE35', 'Age.MALE40',
       'Age.MALE45', 'Age.MALE50', 'Age.MALE55', 'Age.MALE60',
       'Age.MALE65', 'Age.MALE70', 'Age.MALE75', 'Age.MALE80',
       'Age.MALE85', 'Age.FEM0', 'Age.FEM5', 'Age.FEM10', 'Age.FEM15',
       'Age.FEM20', 'Age.FEM25', 'Age.FEM30', 'Age.FEM35', 'Age.FEM40',
       'Age.FEM45', 'Age.FEM50', 'Age.FEM55', 'Age.FEM60', 'Age.FEM65',
       'Age.FEM70', 'Age.FEM75', 'Age.FEM80', 'Age.FEM85'], dtype=object)

In [11]:
# view a sample of the `Age` data collection
df.loc['Age'].head()

analysisVariable                 alias  \
dataCollectionID                                          
Age                     Age.MALE0    2017 Males Age 0-4   
Age                     Age.MALE5    2017 Males Age 5-9   
Age                    Age.MALE10  2017 Males Age 10-14   
Age                    Age.MALE15  2017 Males Age 15-19   
Age                    Age.MALE20  2017 Males Age 20-24   

                                       fieldCategory vintage  
dataCollectionID                                              
Age               2017 Age: 5 Year Increments (Esri)    2017  
Age               2017 Age: 5 Year Increments (Esri)    2017  
Age               2017 Age: 5 Year Increments (Esri)    2017  
Age               2017 Age: 5 Year Increments (Esri)    2017  
Age               2017 Age: 5 Year Increments (Esri)    2017

### Enriching an address

In [23]:
enrich(study_areas=["380 New York St Redlands CA 92373"],  
       data_collections=['Age'])

FEM0  FEM10  FEM15  FEM20  FEM25  FEM30  FEM35  FEM40  FEM45  FEM5  \
0   435    377    417    602    615    577    508    403    386   391   

                         ...                          OBJECTID          X  \
0                        ...                                 1 -117.19567   

           Y                  aggregationMethod    areaType  bufferRadii  \
0  34.056488  BlockApportionment:US.BlockGroups  RingBuffer            1   

   bufferUnits  bufferUnitsAlias  sourceCountry  \
0    esriMiles             Miles             US   

                                               SHAPE  
0  {'spatialReference': {'latestWkid': 4326, 'wki...  

[1 rows x 48 columns]

# Reports

In [12]:
# print a sample of the reports available for USA
usa.reports.head(10)

id                                title  \
0  census2010_profile                  2010 Census Profile   
1         acs_housing                  ACS Housing Summary   
2      acs_population               ACS Population Summary   
3              55plus                      Age 50+ Profile   
4          agesexrace           Age by Sex by Race Profile   
5              agesex                   Age by Sex Profile   
6            cex_auto  Automotive Aftermarket Expenditures   
7        business_loc                     Business Locator   
8    business_summary                     Business Summary   
9   community_profile                    Community Profile   

            categories      formats  
0       [Demographics]  [pdf, xlsx]  
1       [Demographics]  [pdf, xlsx]  
2       [Demographics]  [pdf, xlsx]  
3       [Demographics]  [pdf, xlsx]  
4       [Demographics]  [pdf, xlsx]  
5       [Demographics]  [pdf, xlsx]  
6  [Consumer Spending]  [pdf, xlsx]  
7           [Business]  [pdf, xlsx]  
8           [Business]  [pdf, xlsx]  
9       [Demographics]  [pdf, xlsx]

In [13]:
# total number of reports available
usa.reports.shape

(49, 4)

### Creating Reports

In [14]:
report = create_report(study_areas=["380 New York Street, Redlands, CA"],
                     report="tapestry_profileNEW",
                     export_format="PDF", 
                     out_folder=r"c:\xc", out_name="esri_tapestry_profile.pdf")
report

'c:\\xc\\esri_tapestry_profile.pdf'

## Finding named statistical areas

Each country has several named statistical areas in a hierarchy of geography levels (such as states, counties, zip codes, etc).

In [24]:
%config IPCompleter.greedy=True

In [50]:
usa.subgeographies.states['California'].counties['San_Bernardino_County']

<NamedArea name:"San Bernardino County" area_id="06071", level="US.Counties", country="United States">

In [27]:
usa.subgeographies.states['California'].counties['San_Bernardino_County'].tracts['060710001.03']

<NamedArea name:"060710001.03" area_id="06071000103", level="US.Tracts", country="United States">

In [28]:
usa.subgeographies.states['California'].zip5['92373']

<NamedArea name:"Redlands" area_id="92373", level="US.ZIP5", country="United States">

The named areas can also be drawn on a map, as they include a `geometry` property.

In [29]:
m = gis.map('Redlands, CA', zoomlevel=11)
m

MapView(basemaps=['dark-gray', 'dark-gray-vector', 'gray', 'gray-vector', 'hybrid', 'national-geographic', 'oceans', 'osm', 'satellite', 'streets', 'streets-navigation-vector', 'streets-night-vector', 'streets-relief-vector', 'streets-vector', 'terrain', 'topo', 'topo-vector'], center=[34.05538000000007, -117.18258999999995], zoom=11)

In [30]:
m.draw(usa.subgeographies.states['California'].zip5['92373'].geometry)

# Different geography levels for different country

In [31]:
india = Country.get('India')

In [32]:
india.subgeographies.states['Uttar_Pradesh'].districts['Baghpat'].subdistricts['Baraut']

<NamedArea name:"Baraut" area_id="09080001", level="IN.Subdistricts", country="India">

### Searching for named areas within a country

In [33]:
riversides_in_usa = usa.search('Riverside')
print("number of riversides in the US: " + str(len(riversides_in_usa)))

# list a few of them
riversides_in_usa[:10]

number of riversides in the US: 83


[<NamedArea name:"Riverside" area_id="147435", level="Cities", country="United States">,
 <NamedArea name:"Riverside" area_id="147436", level="Cities", country="United States">,
 <NamedArea name:"Riverside" area_id="147437", level="Cities", country="United States">,
 <NamedArea name:"Riverside" area_id="147438", level="Cities", country="United States">,
 <NamedArea name:"Riverside" area_id="147439", level="Cities", country="United States">,
 <NamedArea name:"Riverside" area_id="147440", level="Cities", country="United States">,
 <NamedArea name:"Riverside" area_id="147441", level="Cities", country="United States">,
 <NamedArea name:"Riverside" area_id="147442", level="Cities", country="United States">,
 <NamedArea name:"Riverside" area_id="147443", level="Cities", country="United States">,
 <NamedArea name:"Riverside" area_id="147444", level="Cities", country="United States">]

For instance, you can make a map of all the riversides in the US

In [34]:
usamap = gis.map('United States', zoomlevel=4)
usamap

MapView(basemaps=['dark-gray', 'dark-gray-vector', 'gray', 'gray-vector', 'hybrid', 'national-geographic', 'oceans', 'osm', 'satellite', 'streets', 'streets-navigation-vector', 'streets-night-vector', 'streets-relief-vector', 'streets-vector', 'terrain', 'topo', 'topo-vector'], center=[39.39870315600007, -99.41461918999994], zoom=4)

In [35]:
for riverside in riversides_in_usa:
    usamap.draw(riverside.geometry)

#### Filtering named areas by geography level

In [30]:
[level['id'] for level in usa.levels]

['US.WholeUSA',
 'US.States',
 'US.DMA',
 'US.CD',
 'US.CBSA',
 'US.Counties',
 'US.CSD',
 'US.ZIP5',
 'US.Places',
 'US.Tracts',
 'US.BlockGroups']

In [29]:
usa.search(query='Riverside', layers=['US.Counties'])

[<NamedArea name:"Riverside County" area_id="06065", level="US.Counties", country="United States">]

## Finding businesses

In [36]:
businesses = find_businesses(search_string="Starbucks",
                          return_geometry=True,
                          spatial_filter={"Locations":["NY,TONAWANDA,14150",
                                                       "NJ, CAMDEN, 08102",
                                                       "KY,LOUISVILLE,40204",
                                                       "WA,SEATTLE,98108"]})
# print a sample
businesses

ADDR        CITY     CONAME  EMPNUM FRNCOD HDBRCH  \
0  5963 CORSON AVE S # A-184     SEATTLE  STARBUCKS      20      4      2   
1                406 PENN ST      CAMDEN  STARBUCKS      15      4      2   
2             972 BAXTER AVE  LOUISVILLE  STARBUCKS      16      4      2   
3                326 PENN ST      CAMDEN  STARBUCKS      12      4      2   
4            4502 12TH AVE S     SEATTLE  STARBUCKS      15      4      2   

  ISCODE     LOCNUM       LOC_NAME     NAICS  \
0         243104437  StreetAddress  72251505   
1         243189941  StreetAddress  72251505   
2         210648473   PointAddress  72251505   
3         683534911  StreetAddress  72251505   
4         714927777   PointAddress  72251505   

                         ...                             SIC     SOURCE  \
0                        ...                          581228  INFOGROUP   
1                        ...                          581228  INFOGROUP   
2                        ...                          581228  INFOGROUP   
3                        ...                          581228  INFOGROUP   
4                        ...                          581228  INFOGROUP   

   SQFTCODE STATE  STATE_NAME STATUS        STREET    ZIP  ZIP4  \
0         B    WA  Washington      M  CORSON AVE S  98108  2619   
1         A    NJ  New Jersey      M       PENN ST  08102  1400   
2         A    KY    Kentucky      M    BAXTER AVE  40204  2064   
3         A    NJ  New Jersey      M       PENN ST  08102  1410   
4         A    WA  Washington      M    12TH AVE S  98108  1805   

                                               SHAPE  
0  {'y': 47.5483999998966, 'spatialReference': {'...  
1  {'y': 39.9480999999659, 'spatialReference': {'...  
2  {'y': 38.2401999999322, 'spatialReference': {'...  
3  {'y': 39.9482999997207, 'spatialReference': {'...  
4  {'y': 47.5633000002997, 'spatialReference': {'...  

[5 rows x 23 columns]

## Study Areas

### Accepted forms of study areas

- **Street address locations** - Locations can be passed as strings of input street addresses, points of interest or place names.
    + **Example:** `"380 New York St, Redlands, CA"`

- **Multiple field input addresses** - Locations described as multiple field input addresses, using dictionaries.
    + **Example:** 
        {"Address" : "380 New York Street",
        "City" : "Redlands",
        "Region" : "CA",
        "Postal" : 92373}    
 
- **Point and line geometries** - Point and line locations, using `arcgis.geometry` instances.
    + **Example Point Location: ** 
    
    `arcgis.geometry.Geometry({"x":-122.435,"y":37.785})`
    
    + ** Example Point location obtained using find_businesses() above: **
     
     `arcgis.geometry.Geometry(businesses.iloc[0]['SHAPE'])`

- **Buffered study areas** - `BufferStudyArea` instances to change the ring buffer size or create drive-time service areas around points specified using one of the above methods. BufferStudyArea allows you to buffer point and street address study areas. They can be created using the following parameters:
        * area: the point geometry or street address (string) study area to be buffered
        * radii: list of distances by which to buffer the study area, eg. [1, 2, 3]
        * units: distance unit, eg. Miles, Kilometers, Minutes (when using drive times/travel_mode)
        * overlap: boolean, uses overlapping rings/network service areas when True, or non-overlapping disks when False
        * travel_mode: None or string, one of the supported travel modes when using network service areas
    + **Example Buffered Location: ** 
    
    `pt = arcgis.geometry.Geometry({"x":-122.435,"y":37.785})
    buffered_area = BufferStudyArea(area=pt, radii=[1,2,3], units="Miles", overlap=False)` 

- **Network service areas** - `BufferStudyArea` also allows you to define drive time service areas around points as well as other advanced service areas such as walking and trucking.
    + **Example: **
    
    `pt = arcgis.geometry.Geometry({"x":-122.435,"y":37.785})
    buffered_area = BufferStudyArea(area=pt, radii=[1,2,3], units="Minutes", travel_mode="Driving")` 

- **Named statistical areas** - 
    + **Example:** 
    
    `usa.subgeographies.states['California'].zip5['92373']`
   
- **Polygon geometries** - Locations can given as polygon geometries.
    + **Example Polygon geometry: ** 
    
    `arcgis.geometry.Geometry({"rings":[[[-117.185412,34.063170],[-122.81,37.81],[-117.200570,34.057196],[-117.185412,34.063170]]],"spatialReference":{"wkid":4326}})`


### Example: Enriching a named statistical area
Enriching zip code 92373 in California using the 'Age' data collection:

In [37]:
redlands = usa.subgeographies.states['California'].zip5['92373']

In [38]:
enrich(study_areas=[redlands], data_collections=['Age'] )

FEM0  FEM10  FEM15  FEM20  FEM25  FEM30  FEM35  FEM40  FEM45  FEM5  \
0   871    931    962   1123   1147   1206   1126   1041   1098   891   

                         ...                          MALE75  MALE80  MALE85  \
0                        ...                             477     317     362   

   OBJECTID  StdGeographyID  StdGeographyLevel  StdGeographyName  \
0         1           92373            US.ZIP5          Redlands   

   aggregationMethod  sourceCountry  \
0      Query:US.ZIP5             US   

                                               SHAPE  
0  {'spatialReference': {'latestWkid': 4326, 'wki...  

[1 rows x 45 columns]

### Example: Enrich all counties in a state

In [39]:
ca_counties = usa.subgeographies.states['California'].counties

In [40]:
counties_df = enrich(study_areas=ca_counties, data_collections=['Age'])
counties_df.head(10)

FEM0  FEM10  FEM15  FEM20  FEM25  FEM30  FEM35  FEM40  FEM45   FEM5  \
0   1039   1027    944    917    965    898    840    795    835   1004   
1   6438   7053   9401  12114   8568   7964   7310   6905   7450   6820   
2    252    264    242    215    215    230    217    241    285    279   
3   2465   2515   2416   2401   2592   2666   2560   2409   2488   2465   
4  20668  18942  17445  17438  18597  16683  14568  13010  12625  19489   
5   3458   3423   3102   3179   3518   3412   3086   2858   2881   3453   
6   2065   2079   1963   1911   2025   1851   1733   1691   1922   2065   
7   4733   5125   5247   5425   5590   5269   4941   4839   5355   4847   
8    371    458    486    426    438    447    390    427    531    440   
9  21466  23726  20863  20830  23275  25123  26114  26147  27054  23197   

                         ...                          MALE75  MALE80  MALE85  \
0                        ...                             349     217     208   
1                        ...                            4059    2744    2883   
2                        ...                             175     104     103   
3                        ...                            1277     746     751   
4                        ...                            3872    2410    2292   
5                        ...                            1142     775     638   
6                        ...                            1009     652     531   
7                        ...                            2882    1770    1710   
8                        ...                             429     244     183   
9                        ...                            9147    6126    6547   

   OBJECTID  StdGeographyID  StdGeographyLevel        StdGeographyName  \
0         1           06021        US.Counties            Glenn County   
1         2           06079        US.Counties  San Luis Obispo County   
2         3           06049        US.Counties            Modoc County   
3         4           06045        US.Counties        Mendocino County   
4         5           06107        US.Counties           Tulare County   
5         6           06101        US.Counties           Sutter County   
6         7           06103        US.Counties           Tehama County   
7         8           06089        US.Counties           Shasta County   
8         9           06063        US.Counties           Plumas County   
9        10           06081        US.Counties        San Mateo County   

   aggregationMethod  sourceCountry  \
0  Query:US.Counties             US   
1  Query:US.Counties             US   
2  Query:US.Counties             US   
3  Query:US.Counties             US   
4  Query:US.Counties             US   
5  Query:US.Counties             US   
6  Query:US.Counties             US   
7  Query:US.Counties             US   
8  Query:US.Counties             US   
9  Query:US.Counties             US   

                                               SHAPE  
0  {'spatialReference': {'latestWkid': 4326, 'wki...  
1  {'spatialReference': {'latestWkid': 4326, 'wki...  
2  {'spatialReference': {'latestWkid': 4326, 'wki...  
3  {'spatialReference': {'latestWkid': 4326, 'wki...  
4  {'spatialReference': {'latestWkid': 4326, 'wki...  
5  {'spatialReference': {'latestWkid': 4326, 'wki...  
6  {'spatialReference': {'latestWkid': 4326, 'wki...  
7  {'spatialReference': {'latestWkid': 4326, 'wki...  
8  {'spatialReference': {'latestWkid': 4326, 'wki...  
9  {'spatialReference': {'latestWkid': 4326, 'wki...  

[10 rows x 45 columns]

In [41]:
m = gis.map('California')
m

MapView(basemaps=['dark-gray', 'dark-gray-vector', 'gray', 'gray-vector', 'hybrid', 'national-geographic', 'oceans', 'osm', 'satellite', 'streets', 'streets-navigation-vector', 'streets-night-vector', 'streets-relief-vector', 'streets-vector', 'terrain', 'topo', 'topo-vector'], center=[0, 0])

In [42]:
lyr = gis.content.import_data(df=counties_df, title="CA county population")

In [43]:
m.add_layer(lyr.layers[0], {'renderer': 'ClassedColorRenderer', 
                            'field_name':'FEM0'})

### Example: Using comparison levels

In [54]:
enrich(study_areas=[redlands], data_collections=['Age'], 
       comparison_levels=['US.Counties', 'US.States'])

c:\xc\github\geosaurus\src\arcgis\features\_data\geodataset\geodataframe.py:1038: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  frame.set_value(index=idx, col=self._geometry_column_name, value=g)


FEM0    FEM10    FEM15    FEM20    FEM25    FEM30    FEM35    FEM40  \
0      871      931      962     1123     1147     1206     1126     1041   
1    84409    83941    82177    84307    91156    83313    76213    71910   
2    79477    76497    74857    82633    90066    79719    70374    65761   
3  1245148  1260883  1269825  1405639  1510193  1429495  1313060  1237282   

     FEM45     FEM5                        ...                          \
0     1098      891                        ...                           
1    71553    84041                        ...                           
2    65744    77343                        ...                           
3  1250328  1251583                        ...                           

   MALE75  MALE80  MALE85  OBJECTID  StdGeographyID  StdGeographyLevel  \
0     477     317     362         1           92373            US.ZIP5   
1   26944   18151   16402         2           06065        US.Counties   
2   17677   10838    9276         3           06071        US.Counties   
3  408593  268305  266546         4              06          US.States   

        StdGeographyName  aggregationMethod  sourceCountry  \
0               Redlands      Query:US.ZIP5             US   
1       Riverside County  Query:US.Counties             US   
2  San Bernardino County  Query:US.Counties             US   
3             California    Query:US.States             US   

                                               SHAPE  
0  {'rings': [[[-117.21461999975689, 34.065140000...  
1  {'spatialReference': {'wkid': 4326, 'latestWki...  
2  {'spatialReference': {'wkid': 4326, 'latestWki...  
3  {'spatialReference': {'wkid': 4326, 'latestWki...  

[4 rows x 45 columns]

### Example: Buffering locations using non overlapping disks 

The example below creates non-overlapping disks of radii 1, 3 and 5 Miles respectively from a street address and enriches these using the 'Age' data collection.

In [44]:
buffered = BufferStudyArea(area='380 New York St Redlands CA 92373',
                           radii=[1,3,5], units='Miles', overlap=False)
enrich(study_areas=[buffered], data_collections=['Age'])

FEM0  FEM10  FEM15  FEM20  FEM25  FEM30  FEM35  FEM40  FEM45  FEM5  \
0   435    377    417    602    615    577    508    403    386   391   
1  1759   1809   2158   2280   2404   2369   2060   1908   1941  1786   
2  2790   2776   2628   3257   3596   2975   2538   2364   2453  2807   

                         ...                          OBJECTID          X  \
0                        ...                                 1 -117.19567   
1                        ...                                 2 -117.19567   
2                        ...                                 3 -117.19567   

           Y                  aggregationMethod         areaType  bufferRadii  \
0  34.056488  BlockApportionment:US.BlockGroups  RingBufferBands            1   
1  34.056488  BlockApportionment:US.BlockGroups  RingBufferBands            3   
2  34.056488  BlockApportionment:US.BlockGroups  RingBufferBands            5   

   bufferUnits  bufferUnitsAlias  sourceCountry  \
0        Miles             Miles             US   
1        Miles             Miles             US   
2        Miles             Miles             US   

                                               SHAPE  
0  {'spatialReference': {'latestWkid': 4326, 'wki...  
1  {'spatialReference': {'latestWkid': 4326, 'wki...  
2  {'spatialReference': {'latestWkid': 4326, 'wki...  

[3 rows x 48 columns]

### Example: Using drive times as study areas
    
The example below creates 5 and 10 minute drive times from a street address and enriches these using the 'Age' data collection.

In [45]:
buffered = BufferStudyArea(area='380 New York St Redlands CA 92373', 
                           radii=[5, 10], units='Minutes', 
                           travel_mode='Driving')
drive_time_df = enrich(study_areas=[buffered], data_collections=['Age'])

In [46]:
drive_time_df

FEM0  FEM10  FEM15  FEM20  FEM25  FEM30  FEM35  FEM40  FEM45  FEM5  \
0   368    318    346    492    513    473    406    325    308   329   
1  2326   2285   2679   3042   3135   2991   2607   2351   2363  2293   

                         ...                          OBJECTID          X  \
0                        ...                                 1 -117.19567   
1                        ...                                 2 -117.19567   

           Y                  aggregationMethod            areaType  \
0  34.056488  BlockApportionment:US.BlockGroups  NetworkServiceArea   
1  34.056488  BlockApportionment:US.BlockGroups  NetworkServiceArea   

   bufferRadii  bufferUnits    bufferUnitsAlias  sourceCountry  \
0            5      Minutes  Drive Time Minutes             US   
1           10      Minutes  Drive Time Minutes             US   

                                               SHAPE  
0  {'spatialReference': {'latestWkid': 4326, 'wki...  
1  {'spatialReference': {'latestWkid': 4326, 'wki...  

[2 rows x 48 columns]

### Visualize results on a map

The returned spatial dataframe can be visualized on a map as shown below:

In [47]:
redlands_map = gis.map('Redlands, CA')
redlands_map.basemap = 'dark-gray-vector'
redlands_map

MapView(basemaps=['dark-gray', 'dark-gray-vector', 'gray', 'gray-vector', 'hybrid', 'national-geographic', 'oceans', 'osm', 'satellite', 'streets', 'streets-navigation-vector', 'streets-night-vector', 'streets-relief-vector', 'streets-vector', 'terrain', 'topo', 'topo-vector'], center=[0, 0])

In [48]:
redlands_map.draw(drive_time_df.to_featureset())

## Saving GeoEnrichment Results

In [48]:
gis.content.import_data(df=drive_time_df, title="Age statistics within 5,10 minutes of drive time from Esri")

<Item title:"Age statistics within 5,10 minutes of drive time from Esri" type:Feature Layer Collection owner:singhqa>